src/main/java/com/mraof/minestuck/event/ServerEventHandler.java

```diff
@@ -142,20 +143,23 @@ public static void onEntityAttack(LivingHurtEvent event)
 	{
 		if(event.getSource().getEntity() != null)
 		{
-			if (event.getSource().getEntity() instanceof ServerPlayer)
+			Entity attacker = event.getSource().getEntity();
+			Entity injured = event.getEntity();
+			
+			if(injured != null)
 			{
-				ServerPlayer player = (ServerPlayer) event.getSource().getEntity();
-				if (event.getEntity() instanceof UnderlingEntity)
-				{    //Increase damage to underling
-					double modifier = PlayerSavedData.getData(player).getEcheladder().getUnderlingDamageModifier();
+				boolean attackerIsRealPlayer = attacker instanceof Player && !(attacker instanceof FakePlayer);
```

Comment:
Should use `ServerPlayer` instead of `Player` to match the cast in `PlayerSavedData.getData((ServerPlayer) attacker)`

Reply:
```suggestion
				boolean attackerIsRealPlayer = attacker instanceof ServerPlayer && !(attacker instanceof FakePlayer);
```